contributor-docs/adrs/adr-XXX-file-structure.md

```diff
@@ -0,0 +1,37 @@
+# ADR XXX: File structure
+
+## Status
+
+Proposed
+
+## Context
+
+Components might be more grokable if they were structured consistently
+
+## Decision
+
+```
+primer-react/
+├─ src/
+│  ├─ BreadCrumbs/
+│  │  ├─ index.ts                    // Just re-exporting?
+│  │  ├─ BreadCrumbs.tsx             // Primary component
+│  │  ├─ BreadCrumbsItem.tsx         // Subcomponent
+│  │  ├─ BreadCrumbs.mdx             // Documentation. Always .mdx, not .md
+│  │  ├─ BreadCrumbs.stories.tsx
+│  │  ├─ BreadCrumbs.test.tsx
+│  │  └─ BreadCrumbs.types.test.tsx
+┆  ┆
+```
+
+Questions I would like to settle in this ADR:
+
+- On subcomponents:
+  - Do subcomponents filenames need to be namespaced with the parent component? (e.g., `BreadCrumbsItem.tsx` vs `Item.tsx`)
+  - Should the `index.ts` in each component directory export every subcomponent under its own name, or should we prefer `Object.assign`ing subcomponents to the parent component?
+  - Should subcomponents always have their own test files? What about type tests and stories?
+- Would this structure help if we wanted to move toward one-package-per-component like react-aria does?
+- Could we set a standard for replacement components? E.g., NewButton vs. Button2
```

Comment:
Ohh interesting — can you say more about why you like that?

I hesitate because it'd mean advancing the status of a component from propsed -> alpha would have to be a breaking change, right?

Reply:
> I hesitate because it'd mean advancing the status of a component from proposed -> alpha would have to be a breaking change, right?

Don't think so. we should use `drafts` for replacement components which can be at a more advanced maturity than their current counterpart. For example, ActionList v1 is alpha but ActionList v2 can be at beta maturity while being in drafts. This is why we chose drafts over other names. More context here: https://github.com/primer/react/pull/1609#issue-1053897027

I like it more because it matches the bundle scope. `drafts` are not part of the main bundle.